### PR TITLE
feat: add `network_option` if it is not present w/ `NetworkBoundCommand` [APE-1575]

### PR DESCRIPTION
**Pull request by rafael-abuawad**
### What I did

`NetworkBoundCommand` automatically adds `network_option` if it is not there.

fixes: #1572

### How I did it

`network_option` could be pass to the default arguments if is not present. WIP

### How to verify it

Haven't had too much time to test it yet, will update PR once tested. 

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Comment by rafael-abuawad** [comment]
I'm commenting the `step` param because the pre-commit was complaining that is was no defined.

File: src/ape/cli/commands.py [hunk 1]
```diff
@@ -23,6 +24,13 @@ class NetworkBoundCommand(click.Command):
     """
 
     def invoke(self, ctx: Context) -> Any:
+        # Check if the network_option is defined, if not, add it with default values
+        if not any(
+            isinstance(param, click.core.Option) and param.name == "network"
+            for param in self.params
+        ):
+            self.params.append(network_option())
```
Comment (antazoey) [inline src/ape/cli/commands.py:32]: Usage should be updated. Every place we are using both `NetworkBoundCommand` and `network_option` in core should be able to be simplified now.

Examples:

1. `clid.md` guide
2. ape-cache CLI tools
3. ape-console CLI tools
4. ape-run CLI tools

and anywhere else you find it

File: src/ape/managers/chain.py [hunk 2]
```diff
@@ -530,10 +530,11 @@ def __getitem_int(self, index: int) -> ReceiptAPI:
 
     @__getitem__.register
     def __getitem_slice(self, indices: slice) -> List[ReceiptAPI]:
-        start, stop, step = (
+        # start, stop, step = (
+        start, stop = (
             indices.start or 0,
             indices.stop or len(self),
-            indices.step or 1,
+            # indices.step  or 1,
```
Comment (antazoey) [inline src/ape/managers/chain.py:537]: Let's just not make this change. I don't know why your pre-commit is telling you to do this...

File: src/ape/cli/commands.py (hunk 1 above)
Comment (antazoey) [inline src/ape/cli/commands.py:32]: I tried this out but it doesn't seem to be working?

**Comment by antazoey** [comment]
closing in favor https://github.com/ApeWorX/ape/pull/1764

**Pull request closed without merging**